doc/explanation/package-management.md

```diff
@@ -0,0 +1,226 @@
+there is no obligation to switch to declaring dependencies in the
+`dune-project`. Likewise the generation of `.opam` files will still work.
+
+## Locking
+
+To go from a project's set of dependency constraints to a set of installed
+packages and versions, there needs to be a step to determine the right packages
+and their versions to be installed.
+
+In `opam`, this process happens as part of `opam install`, which links finding
+a solution that satifies the given constrains and installation into one step.
+Dune on the other hand separates the steps of finding a solution and installing. 
+First a solution is found and then packages are installed.
+
+The idea of finding a solution and recording it for later is popular in other
+programming language package managers like NPM and is usually called locking.
+
+:::{note}
+`opam` also supports creating lock files. However, these are not as central to
+the opam workflow as they are in the case of package management in Dune, which
+always requires a set of locked packages.
+:::
+
+In the most general sense, a package lock is just a set of specific packages and
+their versions to be installed.
+
+A Dune lock file extends this to a directory with files that describe the
+dependencies. It includes the package's name and version. Unlike many
+other package managers, the files include a lot of other information as well,
+such as the location of the source archives to download (since there is no
+central location for all archives), the build instructions (since each package
+can use its own way of building), and additional metadata like the system
+packages it depends upon.
+
+The information is stored in a directory (`dune.lock` by default) as separate
+files, because that makes them easier to manage in source control as it
+leads to fewer potential merge conflicts and simplifies review processes.
+Storing additional files like patches are also more elegant this way.
+
+### Package Repository Management
+
+To find a valid solution that allows a project to be built, it is necessary to
+know what packages exist, what versions of these packages exist, and what other
+packages these depend on, etc.
+
+In opam, this information is tracked in a central repository called
+[`opam-repository`](https://github.com/ocaml/opam-repository), which contains all
+the metadata for published packages.
+
+It is managed using Git, and opam typically uses a snapshot to find the
+dependencies when searching for a solution that satisfies the constraints.
+
+Likewise, Dune uses the same repository; however, instead of file snapshots, 
+it uses the Git repository directly. In fact, Dune maintains a shared
+internal cache containing all Git repositories that projects use. The
+advantage is that updates to the metadata are very fast because only the newest
+revisions have to be retrieved. The downside is that for the creation of the
+cache, the entire repository has to be cloned first.
```

Comment:
It's the metadata repository here: the `opam-repository` one.

Reply:
This is in the "package repository management" section, so I (hoped) it was clear that it is about the opam metadata.

It will normally use the js_of_ocaml release tarball if that's what your lockfile refers to. But if you use a pin to specify the jsoo git repo it will also add that repo to the git revision cache (thus multiple projects doing this will not have to clone jsoo twice).

Reply:
To me, the sentence " In fact, Dune maintains a shared internal cache containing all Git repositories that projects use. " means you're not talking only about opam metadata anymore. You're explicitly generalizing